state/azure/cosmosdb/cosmosdb.go

```diff
@@ -214,20 +203,8 @@ func (c *StateStore) Set(req *state.SetRequest) error {
 		options = append(options, documentdb.ConsistencyLevel(documentdb.Eventual))
 	}
 
-	b, ok := req.Value.([]uint8)
-	if ok {
-		// data arrived in bytes and already json.  Don't marshal the Value field again.
-		item := CosmosItemWithRawMessage{ID: req.Key, Value: b, PartitionKey: partitionKey}
-		var marshalled []byte
-		marshalled, err = convertToJSONWithoutEscapes(item)
```

Comment:
This code was relying on the fact that the cosmosdb APIs handle `[]byte` and special case it. This approach isn't really needed, since we can just use `CosmosItem` for both cases.